cmd/container-manager/manage_ctr_mgr.sh

```diff
@@ -0,0 +1,323 @@
+}
+
+log_message(){
+	message="$1"
+	logger -s -t "$(basename $0)" "$message"
+}
+
+parse_subcommand_options(){
+	while getopts ":fr:t:" opt; do
+		case $opt in
+			f)
+				force=true
+				;;
+			r)
+				runtime="${OPTARG}"
+				;;
+			t)
+				tag="${OPTARG}"
+				;;
+			\?)
+				echo "Invalid option: -${OPTARG}" >&2
+				usage
+				exit 1
+		esac
+	done
+}
+
+# This function handles the installation of the required docker version.
+install_docker(){
+	# Check if docker is present in the system
+	if [ "$(info_docker)" ] && [ ${force} == false ]; then
+		die "Docker is already installed. Please use -f flag to force new installation"
+	elif [ "$(info_docker)" ] && [ ${force} == true ]; then
+		remove_docker
+	fi
+
+	if [ -z "$tag" ] || [ "$tag" == "latest" ] ; then
+		# If no tag is recevied, install latest compatible version
+		log_message "Installing docker $docker_version"
+		docker_version=${docker_version/v}
+		docker_version=${docker_version/-*}
+		if [ "$ID" == "ubuntu" ]; then
+			pkg_name="docker-ce=${docker_version}.0~ce-0~ubuntu"
+			repo_url="https://download.docker.com/linux/ubuntu"
+			sudo -E apt-get -y install apt-transport-https ca-certificates software-properties-common
+			curl -fsSL "${repo_url}/gpg" | sudo apt-key add -
+			sudo -E add-apt-repository "deb [arch=amd64] ${repo_url} $(lsb_release -cs) stable"
+			sudo -E apt-get update
+			sudo -E apt-get -y install docker-ce
+		elif [ "$ID" == "fedora" ]; then
+			pkg_name="docker-ce-${docker_version}.0.ce"
+			repo_url="https://download.docker.com/linux/fedora/docker-ce.repo"
+			sudo -E dnf -y install dnf-plugins-core
+			sudo -E dnf config-manager --add-repo "$repo_url"
+			sudo -E dnf makecache
+			sudo -E dnf -y install "$pkg_name"
+			sudo systemctl start docker
+		fi
+	elif [ "$tag" == "swarm" ]; then
+		# If tag is swarm, install docker 1.12.1
+		log_message "Installing docker $docker_swarm_version"
+		if [ "$ID" == "ubuntu" ]; then
+			# We stick to the xenial repo, since it is the only one that
+			# provides docker 1.12.1
+			pkg_name="docker-engine=${docker_swarm_version}-0~xenial"
+			repo_url="https://apt.dockerproject.org"
+			sudo -E apt-get -y install apt-transport-https ca-certificates
+			curl -fsSL "${repo_url}/gpg" | sudo apt-key add -
+			sudo -E add-apt-repository "deb ${repo_url}/repo ubuntu-xenial main"
+			sudo -E apt-get update
+			sudo -E apt-get -y install --allow-downgrades "$pkg_name"
+		elif [ "$ID" == "fedora" ]; then
+			# We stick to the Fedora 24 repo, since it is the only one that
+			# provides docker 1.12.1
+			pkg_name="docker-engine-${docker_swarm_version}-1.fc24"
+			repo_url="https://yum.dockerproject.org"
+			gpg_key="gpg"
+			sudo -E dnf -y install dnf-plugins-core
+			sudo -E dnf config-manager --add-repo  "${repo_url}/repo/main/fedora/24"
+			curl -O "${repo_url}/${gpg_key}"
+			sudo rpm --import "./${gpg_key}"
+			rm "./${gpg_key}"
+			sudo -E dnf makecache
+			sudo -E dnf -y install "$pkg_name"
+			sudo systemctl start docker
```

Comment:
I'm assuming Fedora doesn't start the service by default by Ubuntu does? It still might be best to move that line to outside the `if` tests and make it `sudo systemctl restart docker` so it would work in all scenarios.